src/extlibs/er_force_sim/src/amun/simulator/simball.cpp

```diff
@@ -69,31 +74,66 @@ SimBall::~SimBall()
     delete m_motionState;
 }
 
-void SimBall::begin()
+void SimBall::begin(bool robot_collision)
 {
     // custom implementation of rolling friction
-    const btVector3 p = m_body->getWorldTransform().getOrigin();
+    const btVector3 p        = m_body->getWorldTransform().getOrigin();
+    const btVector3 velocity = m_body->getLinearVelocity();
     if (p.z() < BALL_RADIUS * 1.1 * SIMULATOR_SCALE)
     {  // ball is on the ground
-        const btVector3 velocity = m_body->getLinearVelocity();
-        if (velocity.length() < 0.01 * SIMULATOR_SCALE)
+        bool is_stationary = velocity.length() < STATIONARY_BALL_SPEED * SIMULATOR_SCALE;
+        bool should_roll   = velocity.length() < rolling_speed * SIMULATOR_SCALE;
```

Comment:
can rolling_speed be a constant like STATIONARY_BALL_SPEED

Reply:
rolling speed is a variable that depends on the initial velocity, it gets updated whenever the ball starts sliding (see simball.cpp::118)